src/main/java/gomedic/logic/parser/ProfileCommandParser.java

```diff
@@ -0,0 +1,52 @@
+package gomedic.logic.parser;
+
+import java.util.stream.Stream;
+
+import gomedic.commons.core.Messages;
+import gomedic.logic.commands.ProfileCommand;
+import gomedic.logic.parser.exceptions.ParseException;
+import gomedic.model.activity.Description;
+import gomedic.model.commonfield.Name;
+
+/**
+ * Parses input arguments and creates a new ProfileCommand object.
+ */
+public class ProfileCommandParser implements Parser<ProfileCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the ProfileCommand
+     * and returns a ProfileCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ProfileCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(
+                        args,
+                        CliSyntax.PREFIX_NAME,
+                        CliSyntax.PREFIX_DESCRIPTION);
+
+        if (!arePrefixesPresent(argMultimap,
+                CliSyntax.PREFIX_NAME,
+                CliSyntax.PREFIX_DESCRIPTION)
```

Comment:
Similarly here, maybe try to decompose the three fields

Reply:
Actually the other rationale, is that the fields would be used for the referral template, so description would not be of much use for the referral template also